src/chunkserver/network_worker_thread.h

```diff
@@ -168,14 +168,206 @@ struct ChunkserverEntry {
 	ChunkserverEntry(ChunkserverEntry &&) = delete;
 	ChunkserverEntry &operator=(ChunkserverEntry &&) = delete;
 
-	/// Destructor: releases the sockets and packet resources.
+	/// Destructor: closes the sockets.
 	~ChunkserverEntry();
 
 	/// Attaches a packet to the output packet list (taking ownership).
 	inline void attachPacket(std::unique_ptr<PacketStruct> &&packet);
+
 	/// Preserves the inputPacket buffer into writePacket (to avoid copying it).
 	/// Used for write operations, where the data comes from the network.
 	inline void preserveInputPacket();
+
+	/// Creates an attached packet from the given vector.
+	/// The function takes ownership of the vector.
+	void createAttachedPacket(std::vector<uint8_t> &packet);
+
+	/// Creates an attached packet with the given type and operation size.
+	///
+	/// @param type The type of the packet.
+	/// @param opSize The size of the operation.
+	/// @return Pointer to the created packet data.
+	uint8_t *createAttachedPacket(uint32_t type, uint32_t opSize);
```

Comment:
```suggestion
	const uint8_t *createAttachedPacket(uint32_t type, uint32_t opSize);
```
Consider returning const pointer to prevent caller from accidentally sending the wrong data.

Reply:
This implies changes in other files like the hddspacemgr, which are not related to this change. Also at the end the returned buffer is serialized, needing a const_cast which is also not recommended.